src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+    SpriteRenderer_LoadAnimResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 78, FALSE, 55512);
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 75, FALSE, 1, NNS_G2D_VRAM_TYPE_2DMAIN, 55512);
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 75, FALSE, 1, NNS_G2D_VRAM_TYPE_2DMAIN, 55513);
+    SpriteRenderer_LoadCharResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 76, FALSE, NNS_G2D_VRAM_TYPE_2DMAIN, 55513);
+    SpriteRenderer_LoadCellResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 77, FALSE, 55513);
+    SpriteRenderer_LoadAnimResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 78, FALSE, 55513);
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 75, FALSE, 1, NNS_G2D_VRAM_TYPE_2DMAIN, 55514);
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(data->spriteRenderer, data->spriteGfxHandler, narc, 75, FALSE, 1, NNS_G2D_VRAM_TYPE_2DMAIN, 55515);
+    NARC_Delete(narc);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_UnloadSpriteGfx(RegisterHallOfFameData *data) {
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55512);
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55513);
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55514);
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55515);
+    SpriteGfxHandler_UnloadCharObjById(data->spriteGfxHandler, 55512);
+    SpriteGfxHandler_UnloadCellObjById(data->spriteGfxHandler, 55512);
+    SpriteGfxHandler_UnloadAnimObjById(data->spriteGfxHandler, 55512);
+    SpriteGfxHandler_UnloadCharObjById(data->spriteGfxHandler, 55513);
+    SpriteGfxHandler_UnloadCellObjById(data->spriteGfxHandler, 55513);
+    SpriteGfxHandler_UnloadAnimObjById(data->spriteGfxHandler, 55513);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_SetPicGfxAndPltt(RegisterHallOfFameData *data, RegisterHofMon *hofMon, u8 whichFacing, int picIdx) {
+    SomeDrawPokemonStruct drawMonStruct;
+    GetPokemonSpriteCharAndPlttNarcIds(&drawMonStruct, hofMon->mon, whichFacing);
+    RegisterHallOfFame_ReplaceSpriteChar(
+        whichFacing == MON_PIC_FACING_FRONT ? hofMon->frontspriteCharbuf : hofMon->backspriteCharbuf,
+        NNS_G2dGetImageLocation(Sprite_GetImageProxy(data->monPics[picIdx]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN),
+        3200
+    );
+    GfGfxLoader_GXLoadPal(
+        (NarcId)drawMonStruct.narcID,
+        drawMonStruct.palDataID,
+        GF_PAL_LOCATION_MAIN_OBJ,
+        (enum GFPalSlotOffset)NNS_G2dGetImagePaletteLocation(Sprite_GetPaletteProxy(data->monPics[picIdx]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN),
+        0x20,
+        HEAP_ID_REGISTER_HALL_OF_FAME
+    );
+}
+
+int ov63_dummy_00(RegisterHallOfFameData *data);
```

Comment:
should this one be static?

Reply:
iirc i tried to make it static, and it yeeted the data reference creating a shift